internal/operators/metallb/metallb_operator.go

```diff
@@ -0,0 +1,173 @@
+		}
+		return result, nil
+	}
+
+	return result, nil
+}
+
+// ValidateHost validates host requirements for MetalLB
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host, additionalOperatorRequirements *models.ClusterHostRequirementsDetails) (api.ValidationResult, error) {
+	if host.Inventory == "" {
+		message := "Missing Inventory in the host"
+		return api.ValidationResult{Status: api.Pending, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, nil
+	}
+
+	// MetalLB doesn't have specific hardware requirements beyond basic cluster requirements
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetHostValidationID()}, nil
+}
+
+// GenerateManifests generates manifests for the operator
+func (o *operator) GenerateManifests(cluster *common.Cluster) (map[string][]byte, []byte, error) {
+	return Manifests(cluster)
+}
+
+// GetProperties provides description of operator properties: none required
+func (o *operator) GetProperties() models.OperatorProperties {
+	return models.OperatorProperties{}
+}
+
+// GetMonitoredOperator returns MonitoredOperator corresponding to MetalLB
+func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
+	return &Operator
+}
+
+// GetHostRequirements provides operator's requirements towards the host
+func (o *operator) GetHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	// MetalLB has minimal resource requirements
+	return &models.ClusterHostRequirementsDetails{
+		CPUCores: o.Config.MetalLBCPUPerHost,
+		RAMMib:   o.Config.MetalLBMemoryPerHostMiB,
+	}, nil
+}
+
+// GetPreflightRequirements returns operator hardware requirements that can be determined with cluster data only
+func (o *operator) GetPreflightRequirements(ctx context.Context, cluster *common.Cluster) (*models.OperatorHardwareRequirements, error) {
+	dependencies, err := o.GetDependencies(cluster)
+	if err != nil {
+		return &models.OperatorHardwareRequirements{}, err
+	}
+
+	return &models.OperatorHardwareRequirements{
+		OperatorName: o.GetName(),
+		Dependencies: dependencies,
+		Requirements: &models.HostTypeHardwareRequirementsWrapper{
+			Master: &models.HostTypeHardwareRequirements{
+				Qualitative: []string{},
+				Quantitative: &models.ClusterHostRequirementsDetails{
+					CPUCores: o.Config.MetalLBCPUPerHost,
+					RAMMib:   o.Config.MetalLBMemoryPerHostMiB,
+				},
+			},
+			Worker: &models.HostTypeHardwareRequirements{
+				Qualitative: []string{},
+				Quantitative: &models.ClusterHostRequirementsDetails{
+					CPUCores: o.Config.MetalLBCPUPerHost,
+					RAMMib:   o.Config.MetalLBMemoryPerHostMiB,
+				},
+			},
+		},
+	}, nil
+}
+
+// GetSupportedArchitectures returns supported architectures for MetalLB
+func (o *operator) GetSupportedArchitectures() []string {
+	return []string{
+		common.X86CPUArchitecture,
+		common.ARM64CPUArchitecture,
+	}
+}
```

Comment:
I don't see this function is used (maybe I am missing), but if it is it the validation which uses it should move to `featuresupport`